script.js

```diff
@@ -1,4 +1,210 @@
+
+// Set current game mode so user will roll dice when click submit
+var currentGameMode = player1Roll;
+
+//create function for dice roll
+var diceRoll = function () {
+  return Math.floor(Math.random() * 6) + 1;
+};
+
+//create function emcompassing all game modes & game logic
+var beatThatGame = function (input) {
+  // game mode: player 1 to roll
+  if (currentGameMode == player1Roll) {
+    if (input != "") {
+      myOutputValue = "Please roll the dice by clicking 'submit'";
+      return myOutputValue;
+    }
+    player1FirstRollOutput = diceRoll();
+    player1SecondRollOutput = diceRoll();
+    player1Array.push(player1FirstRollOutput);
+    console.log("pushed player1firstrolloutput into player1array");
+
+    player1Array.push(player1SecondRollOutput);
+    console.log("pushed player1secondrolloutput into player1array");
+
+    console.log(player1Array);
+
+    currentGameMode = player1Select; //change game mode as soon as dices are rolled so it won't re-roll
+    console.log("game mode is changed to player1select");
+
+    myOutputValue = `<b>WELCOME PLAYER 1 </b>🎲🎲🎲 <br><br>You rolled ${player1FirstRollOutput} for Dice 1 and ${player1SecondRollOutput} for Dice 2.<br><br>${selectDiceOrderText} `;
+    return myOutputValue;
+  }
+
+  //game mode: player 1 to select which dice first
+  if (currentGameMode == player1Select) {
+    // how to collect user choice of which dice goes first? (kiv)
+    userInput = Number(input);
+    var player2TurnText = "It is now Player 2's turn. Click submit to proceed.";
+
+    //input validation for (1) nan inputs (2) not 1 and not 2
+
+    //for (1) nan inputs
+    if (Number.isNaN(userInput) == true) {
+      myOutputValue = "sorry, please enter a number.";
+      return myOutputValue;
+    }
+
+    if (userInput != 1 && userInput != 2) {
+      myOutputValue = "Please insert only either '1' or '2'.";
+      return myOutputValue;
+    }
+
+    if (userInput == 1) {
+      //if input is 1, let the first number go first. Else if input is 2, let second number go first
+
+      player1Number = String(player1Array[0]) + String(player1Array[1]);
+      myOutputValue = `Player 1, you chose Dice ${userInput} first.<br><br>You rolled ${player1Array}.<br><b>Your number is ${player1Number}.</b><br><br>${player2TurnText}`;
+      currentGameMode = player2Roll; // change game mode should be within conditionals
+      console.log("game mode is changed to player2roll");
+      return myOutputValue;
+    } else if (userInput == 2) {
+      player1Number = String(player1Array[1]) + String(player1Array[0]);
+      myOutputValue = `Player 1, you chose Dice ${userInput} first.<br><br>You rolled ${player1Array}.<br><b>Your number is ${player1Number}.</b><br><br>${player2TurnText} `;
+      currentGameMode = player2Roll; // change game mode should be within conditionals
+      console.log("game mode is changed to player2roll");
+
+      return myOutputValue;
+    }
```

Comment:
Good way to calculate the score of the user. But notice how your lines 105-108 and lines 111-115 are repeated and doing the same thing? In programming, we try not to repeat ourselves if possible - the reason for this is 1. for readability purpose, and 2. for code maintainability purpose, imagine you notice a bug here and need to change something, then you need to change it twice for the code sections you repeated. And this gets more complex if there are more instances of unnecessary repetition in your code. 

Consider having an if-else that calculates the `player1Number` based on the desired order, then doing the other stuff after the if-else.

```
    if (userInput == 1) {
      //if input is 1, let the first number go first. Else if input is 2, let second number go first
      player1Number = String(player1Array[0]) + String(player1Array[1]);
    } else if (userInput == 2) {
      player1Number = String(player1Array[1]) + String(player1Array[0]);
    }

      myOutputValue = `Player 1, you chose Dice ${userInput} first.<br><br>You rolled ${player1Array}.<br><b>Your number is ${player1Number}.</b><br><br>${player2TurnText}`;
      currentGameMode = player2Roll; // change game mode should be within conditionals
      console.log("game mode is changed to player2roll");
      return myOutputValue;
```

Reply:
also consider a case where the user rolled the same number for both die, does it make sense from a game and user pov to ask them for their desired order? Multiple ways to incorporate this!